applications/ChimeraApplication/custom_processes/apply_chimera_process.h

```diff
@@ -0,0 +1,630 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+// ==============================================================================
+//  ChimeraApplication
+//
+//  License:         BSD License
+//                   license: ChimeraApplication/license.txt
+//
+//  Authors:        Aditya Ghantasala, https://github.com/adityaghantasala
+// 					Navaneeth K Narayanan
+//					Rishith Ellath Meethal
+// ==============================================================================
+//
+#if !defined(KRATOS_APPLY_CHIMERA_H_INCLUDED)
+#define KRATOS_APPLY_CHIMERA_H_INCLUDED
+
+// System includes
+#include <algorithm>
+#include <numeric>
+#include <unordered_map>
+#include "omp.h"
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "processes/process.h"
+#include "includes/model_part.h"
+#include "containers/model.h"
+#include "includes/variables.h"
+#include "includes/linear_master_slave_constraint.h"
+#include "processes/calculate_distance_to_skin_process.h"
+#include "input_output/vtk_output.h"
+#include "utilities/binbased_fast_point_locator.h"
+#include "factories/standard_linear_solver_factory.h"
+#include "processes/variational_distance_calculation_process.h"
+#include "utilities/parallel_levelset_distance_calculator.h"
+#include "utilities/builtin_timer.h"
+#include "utilities/variable_utils.h"
+
+// Application includes
+#include "chimera_application_variables.h"
+#include "custom_utilities/hole_cutting_utility.h"
+#include "calculate_signed_distance_to_2d_condition_skin_process.h"
+#include "processes/calculate_signed_distance_to_3d_condition_skin_process.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+template <int TDim, class TSparseSpaceType, class TLocalSpaceType>
```

Comment:
It appears you are not using the template parameters TSparseSpaceType and TLocalSpaceType.
```suggestion
template <int TDim>
```